bin/cli.js

```diff
@@ -1,8 +1,88 @@
 #!/usr/bin/env node
 'use strict';
 
-require('codemod-cli').runTransform(
-  __dirname,
-  process.argv[2] /* transform name */,
-  process.argv.slice(3) /* paths or globs */
-);
+function codemodCli() {
+  require('codemod-cli').runTransform(
+    __dirname,
+    process.argv[2] /* transform name */,
+    process.argv.slice(3) /* paths or globs */
+  );
+}
+
+// codemod-cli accepts transforms as either a bare name (e.g. "stage-1-foo") or
+// as a path to the transform folder (which must be either absolute or
+// explicitly relative, e.g. "./transforms/stage-1-foo").
+const STAGE_PATTERN = /(?:(?:^\.)?\/transforms)?stage-(\d+)[^/]*(?:\/$)?/;
+
+switch (STAGE_PATTERN.exec(process.argv[2])?.[1]) {
+  case '2':
+    {
```

Comment:
At the point when we have this many things going on, let's extract functions. Also, please move *all* these `require` statements to top level. 😅 (Among other thing, if/when we can switch this over to using ES modules instead of CommonJS modules, that'll make it *far* easier.)

Reply:
Yeah, I was trying to (a) keep the various code paths from requiring modules they don't need and (b) make it clear what was being used where. The former is kind of a weak argument, though, and you make a good point about ES modules.

That said, the only overlap between the stage 2 handler's requires and the stage 3 handler's is the filename constant. If I'm breaking them out into functions (good idea, by the way) anyway, how do you feel about separate `.js` files for them? The requires could then be moved to the top level, but there'd still be no potential confusion as to where they're being used.

Reply:
In general, I would avoid separate modules unless there's a ton going on. It's just needless jumping around in the editor and in GitHub. There's no paratcular reason for them not to just be `function stage3Transform() { ... }` etc. Mind, that's a pretty mild opinion; if you find it helpful to break out, that's also fine.